### PR TITLE
Add Helm chart for Flux and Helm Operator

### PR DESCRIPTION
**Pull request by stefanprodan**
This PR implements a Helm chart for Flux and optionally the Flux Helm Operator.

```bash
$ helm install --name flux \
--set git.url=git@github.com:weaveworks/flux-helm-test \
--set helmOperator.create=true \
--namespace flux \
./chart/flux
```

More options are available, check the chart readme.

### CODE REVIEW

File: chart/flux/values.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+git:
+  # URL of git repo with Kubernetes manifests; e.g. git@github.com:weaveworks/flux-example
+  url: ""
+  # Branch of git repo to use for Kubernetes manifests
+  branch: "master"
+  # Path within git repo to locate Kubernetes manifests (relative path)
+  path: ""
+  # Username to use as git committer
+  user: "Weave Flux"
```
[the remark this answers is not in the document]
Comment (stefanprodan) [reply chart/flux/values.yaml]: These defaults are hidden in Flux code, I think it's easier for a user to see the defaults in the readme and change them, especially the git user and email.

**Comment by stefanprodan** [comment]
The Helm repo is online at: https://weaveworks.github.io/flux/

Install instructions:

```
helm repo add weaveworks https://weaveworks.github.io/flux

helm install --name flux \
--set git.url=git@github.com:weaveworks/flux-helm-test \
--set helmOperator.create=true \
--namespace flux \
weaveworks/flux

kubectl -n flux logs deployment/flux | grep identity.pub
```

File: chart/flux/README.md [hunk 2]
```diff
@@ -85,12 +85,12 @@ The following tables lists the configurable parameters of the Weave Flux chart a
...
 | `image.repository` | Image repository | `quay.io/weaveworks/flux` 
 | `image.tag` | Image tag | `1.3.1` 
-| `image.pullPoliwell cy` | Image pull policy | `IfNotPresent` 
+| `image.pullPolicy` | Image pull policy | `IfNotPresent` 
 | `resources` | CPU/memory resource requests/limits | None 
 | `rbac.create` | If `true`, create and use RBAC resources | `true`
 | `serviceAccount.create` | If `true`, create a new service account | `true`
 | `serviceAccount.name` | Service account to be used | `flux`
-| `service.type` | Service type to be used | `ClusterIP`
+| `service.type` | Service type to be used (exposing Flux API outside of th cluster is not advise) | `ClusterIP`
```
Comment (squaremo) [inline chart/flux/README.md]: "the"

File: chart/flux/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (squaremo) [reply chart/flux/values.yaml]: OK. It just means it's harder to change the defaults, if we ever want to do that.

**Pull request merged**